integration/test/basic/basic_test.go

```diff
@@ -5,13 +5,54 @@ package basic
 import (
 	"context"
 	"testing"
+	"time"
+
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/microerror"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func TestHelm(t *testing.T) {
+// TestBasic ensures that there is a ready cluster-autoscaler deployment.
+func TestBasic(t *testing.T) {
+	ctx := context.Background()
+	var err error
+
+	// Check metrics-server deployment is ready.
+	err = checkReadyDeployment(ctx)
+	if err != nil {
+		t.Fatalf("could not get metrics: %v", err)
+	}
+}
+
+func checkReadyDeployment(ctx context.Context) error {
 	var err error
 
-	err = ba.Test(context.Background())
+	l.LogCtx(ctx, "level", "debug", "message", "waiting for ready deployment")
```

Comment:
just wondering: should we use `t.Log()`/`t.Logf()` in tests instead of our logger?

Reply:
Interesting idea. I'm not sure. We need the logger anyway because the apptest library uses it. But using `t.Log` is maybe a more native way to do it.

I'll try that out when upgrading one of the next apps.

Reply:
Yeah, we need it to build objects but I meant for actually logging from the test itself. Also, for next tests, it would be cool to use the controller client instead of the `k8sclient`.

Reply:
> Yeah, we need it to build objects but I meant for actually logging from the test itself. Also, for next tests, it would be cool to use the controller client instead of the k8sclient.

Cool I'll try it out for the test logging.

On using the controller client I totally get that for replacing the G8sClient since it has our types.

I'm less sure about using it for K8s resources. But this is also because I'm just less used to using it. But I see the consistency benefit of using the same client for all resources.